apps/navstar_rest/src/navstar_rest_dns_handler.erl

```diff
@@ -0,0 +1,170 @@
+%%%-------------------------------------------------------------------
+%%% @author dgoel
+%%% @copyright (C) 2016, <COMPANY>
+%%% @doc
+%%%
+%%% @end
+%%% Created : 15. Dec 2016 3:25 PM
+%%%-------------------------------------------------------------------
+-module(navstar_rest_dns_handler).
+-author("dgoel").
+
+-include("navstar_rest.hrl").
+-include_lib("dns/include/dns.hrl").
+
+%% API
+-export([init/2]).
+-export([content_types_provided/2, allowed_methods/2, content_types_accepted/2]).
+-export([to_json/2, update_zone/2]).
+
+-define(RECORDS_FIELD, {records, riak_dt_orswot}).
+
+init(Req, Opts) ->
+    {cowboy_rest, Req, Opts}.
+
+content_types_provided(Req, State) ->
+    {[
+        {<<"application/json">>, to_json}
+    ], Req, State}.
+
+allowed_methods(Req, State) ->
+    {[<<"GET">>, <<"POST">>], Req, State}.
+
+content_types_accepted(Req, State) ->
+    {[
+        {<<"application/json">>, update_zone}
+    ], Req, State}.
+
+%% POST handler
+-spec(update_zone(Req::cowboy_req:req(), State::any()) -> {<<>>, Req2::cowboy_req:req(), State::any()}).
+update_zone(Req, State) ->
+    <<"POST">> = cowboy_req:method(Req),
+    true = cowboy_req:has_body(Req),
+    [ZoneName] = cowboy_req:path_info(Req),
+    {ok, Body, _Req2} = cowboy_req:body(Req),
+    Result = case cowboy_req:header(<<"clock">>, Req, <<>>) of
+               <<>> ->
+                   update_zone(ZoneName, [], Body);
+               Clock0 ->
+                   Clock1 = binary_to_term(base64:decode(Clock0)),
+                   update_zone(ZoneName, Clock1, Body)
+               end,
+    Req2 = case Result of
+             {error, Reason} ->
+                 cowboy_req:reply(400, [], Reason, Req);
+             _ ->
+                 cowboy_req:reply(200, Req)
+           end,
+    {<<>>, Req2, State}.
+
+-spec(update_zone(ZoneName::binary(), riak_dt_vclock:vclock(), Body::binary()) ->
+      {ok, no_change} | {error, Reason::term()} | {ok, riak_dt_map:value()}).
+update_zone(ZoneName, Clock, Body0) ->
+    Body1 = jsx:decode(Body0, [return_maps, {labels, atom}]),
```

Comment:
Do we need to use {labels, atoms}? Can the map_to_record function safely convert?